Add a Symfony2 bundle to integrate with the Symfony2 Dependency Injection Container

igorw opened the pull request:
By having the bootstrap managed by the DIC, it becomes a lot easier to use. Also, the XML file is way more maintainable than the annoying manual construction of the object graph.

Sample usage:

```
$collectionBuilder = $container->get('simple_php_easy_plus.number_collection_builder');
$engine = $container->get('simple_php_easy_plus.engine');
$runner = $container->get('simple_php_easy_plus.runner');

$collectionBuilder->add('1');
$collectionBuilder->add('2');

$calcul = new Calcul($engine, $collectionBuilder->resolve());
$runner->run($calcul);

$value = $calcul->getResult()->getValue();
```

This PR depends on #13.

igorw committed:
Add a Symfony2 bundle to integrate with the Symfony2 Dependency Injection Container

By having the bootstrap managed by the DIC, it becomes a lot easier to use.
Also, the XML file is way more maintainable than the annoying manual
construction of the object graph.

Sample usage:

    $collectionBuilder = $container->get('simple_php_easy_plus.number_collection_builder');
    $engine = $container->get('simple_php_easy_plus.engine');
    $runner = $container->get('simple_php_easy_plus.runner');

    $collectionBuilder->add('1');
    $collectionBuilder->add('2');

    $calcul = new Calcul($engine, $collectionBuilder->resolve());
    $runner->run($calcul);

    $value = $calcul->getResult()->getValue();

igorw committed:
Merge branch 'number-collection-builder' into symfony-bundle

* number-collection-builder:
  Mock SimpleNumberStringParserInterface instead of implementation

felds commented:
:+1:

Herzult commented:
This PR contains both bundle & number collection builder.

igorw commented:
Since the NumberCollectionBuilder PR was merged, this now only contains the bundle.

igorw committed:
Move extension creation in extension test

magnusnordlander commented on src/SimplePHPEasyPlus/Bundle/Resources/config/services.xml:
```diff
@@ -0,0 +1,46 @@
+    </parameters>
+
+    <services>
+        <service id="simple_php_easy_plus.number_collection_builder" class="%simple_php_easy_plus.number_collection_builder.class%">
```
Shouldn't this be in the prototype scope?

igorw replied:
Good point, fixed!

igorw committed:
Change number_collection_builder to prototype scope

This ensures that a new builder is created every time. Thanks to
@magnusnordlander for pointing this out.

igorw commented:
Bump.
